waspc/src/Wasp/Generator/WebAppGenerator.hs

```diff
@@ -113,14 +113,21 @@ genPublicDir :: AppSpec -> Generator [FileDraft]
 genPublicDir spec = do
   publicIndexHtmlFd <- genPublicIndexHtml spec
   return $
-    C.mkTmplFd (C.asTmplFile [relfile|public/favicon.ico|]) :
-    publicIndexHtmlFd :
-    ( let tmplData = object ["appName" .= (fst (getApp spec) :: String)]
+    [ publicIndexHtmlFd,
+      genFaviconFd,
+      genManifestFd
+    ]
+      ++ genGoogleSigninImage
+  where
+    maybeAuth = AS.App.auth $ snd $ getApp spec
+    genFaviconFd = C.mkTmplFd (C.asTmplFile [relfile|public/favicon.ico|])
+    genGoogleSigninImage = do
+      let tmplFd = C.mkTmplFd (C.asTmplFile [relfile|public/images/btn_google_signin_dark_normal_web@2x.png|])
+       in if (AS.App.Auth.isGoogleAuthEnabled <$> maybeAuth) == Just True then [tmplFd] else []
```

Comment:
This is not something we should address in this PR, but looking at these checks got me thinking...

We're currently grouping stuff by their targets (public dir, auth dir), which mostly corresponds their purposes (stuff for google, stuff for auth, etc), but not in this case. When Google is enabled, we have to generate files in several (at least two) target locations.

Anyway, since the generation of possibly spread-out-but-related files depends on the same conditions, it might make more sense to group them based on where they're coming from and why (e.g., `genGoogleStuff`). If we did this, we wouldn't have to check the same condition multiple times in different places.

On the other hand, we'll lose one nice property of the current system - the directory tree will no longer directly map to our code.

Reply:
Yeah, this is a good general point for us to consider refactoring across the board in the future. We could have the logical generation entity produce _all_ the files they needed, both server and client, but yeah we then do lose the easily visible directory structure of the current generation setup, which is also nice. 🤔

Reply:
The thing is, I am pretty sure there will be things that depend on multiple conditions. And then you just won't be able to group them in a good way. I don't think by inverting the deps here we will get a better situation.
At least this setup we have right now is somewhat intuitive, since it follows how people think -> Aha, I have auth. we are using google so we should have this button in here. ... -> from bigger things to smaller ones. If we tried to reverse that, I think it wouldn't be so easy to follow.